config/recipes/elastic-agent/ksm-sharding.yaml

```diff
@@ -0,0 +1,471 @@
+apiVersion: agent.k8s.elastic.co/v1alpha1
+kind: Agent
+metadata:
+  name: elastic-agent
+spec:
+  version: 8.10.4
+  elasticsearchRefs:
+  - name: elasticsearch
+  statefulSet:
+    podTemplate:
+      metadata:
+        labels:
+          app.kubernetes.io/name: kube-state-metrics
+      spec:
+        automountServiceAccountToken: true
+        serviceAccountName: elastic-agent
+        containers:
+        - image: registry.k8s.io/kube-state-metrics/kube-state-metrics:v2.10.0
+          livenessProbe:
+            httpGet:
+              path: /healthz
+              port: 8080
+            initialDelaySeconds: 5
+            timeoutSeconds: 5
+          name: kube-state-metrics
+          args:
+            - --pod=$(POD_NAME)
+            - --pod-namespace=$(POD_NAMESPACE)
+          env:
+            - name: POD_NAME
+              valueFrom:
+                fieldRef:
+                  fieldPath: metadata.name
+            - name: POD_NAMESPACE
+              valueFrom:
+                fieldRef:
+                  fieldPath: metadata.namespace
+          ports:
+            - containerPort: 8080
+              name: http-metrics
+            - containerPort: 8081
+              name: telemetry
+          readinessProbe:
+            httpGet:
+              path: /
+              port: 8081
+            initialDelaySeconds: 5
+            timeoutSeconds: 5
+          securityContext:
+            allowPrivilegeEscalation: false
+            capabilities:
+              drop:
+                - ALL
+            readOnlyRootFilesystem: true
+            runAsNonRoot: true
+            runAsUser: 65534
+            seccompProfile:
+              type: RuntimeDefault
+        - name: agent
+          securityContext:
+            runAsUser: 0
+          env:
+          - name: NODE_NAME
+            valueFrom:
+              fieldRef:
+                fieldPath: spec.nodeName
+          - name: POD_NAME
+            valueFrom:
+              fieldRef:
+                fieldPath: metadata.name
+  config:
+    id: 488e0b80-3634-11eb-8208-57893829af4e
+    revision: 2
```

Comment:
Can we remove these 2 lines? If not, maybe add a comment so our users know how to handle them?

Reply:
this came from an existing agent recipe where I drew inspiration of [here](https://github.com/elastic/cloud-on-k8s/blob/main/config/recipes/elastic-agent/kubernetes-integration.yaml#L24). `revision` is not mandatory and can be removed and the same goes for `id` I think

Reply:
ok I tested locally without these two fields and everything works as expected so I remove them